### DISCUSSION TITLE
How to reset the original background color via escape sequences

### DISCUSSION THREAD

**Discussion by Archomeda** [microsoft/terminal#17996] [Q&A]
I've been looking into using a different background color while a command is running. My specific use case is to log into a remote server via SSH, and while I'm logged in, have a different background color.

I got that mostly working. I've written a function that resides in my powershell profile file.
It's as easy as:

```ps
function Open-Remote {
    param(
        $HostName
    )

    Write-Output "`e]4;262;rgb:68/4a/00`e\"
    ssh $HostName
}
```

This nicely changes the background to a dark yellow color.
However, I'm currently struggling to reset the color to its original color. I've found information that it's apparently possible to query the current active color, store it, and change it back when you're done, if I understood this correctly: https://github.com/microsoft/terminal/issues/13091.
I just don't know how to write this in powershell.

Does anyone have an idea?

**Comment by lhecker** [comment]
I believe what you need is #3719.
A few weeks ago, we've received #17879 from j4james which resets the colors when emitting a RIS (``"`ec"``), but that probably won't help you since it would also erase your scrollback.

**Comment by j4james** [comment]
Another option would be to query the palette value with something like ``"`e]4;262;?`e\"``, save the returned OSC sequence, and then echo that string out again when you want to restore the color.

I don't how you'd do that in PowerShell though. I was expecting something like this:
```pwsh
$save = Read-Host -Prompt "`e]4;262;?`e\"
```
But you need to be able to tell `Read-Host` not to echo the input, and also to stop when it reaches a certain character (the final `\`), or maybe a maximum length. I don't know if that's possible - I know almost nothing about PowerShell.

It's also worth mentioning that the palette query support was only just added (PR #17729), so you'd need to be using Windows Terminal Preview.

**Comment by gabrlelg0mes** [comment]
where you color RGB?
